File: src/lec15/Lec15Map.java
package lec15;

import java.util.HashMap;
import java.util.Map;

public class Lec15Map {
    public static void main(String[] args) {

        // JDK 8까지
        Map<Integer, String> map = new HashMap<>();
        map.put(3, "A");
        map.put(4, "B");

        // JDK 9부터
        Map<Integer, String> map2 = Map.of(1, "A", 2, "B");

        // key만 가져오기
        for (int key : map.keySet()) {
            System.out.println(key);
            System.out.println(map.get(key));
        }

        // key-value 쌍으로 가져오기
        for (Map.Entry<Integer, String> entry : map.entrySet()) {
            System.out.println(entry.getKey());
            System.out.println(entry.getValue());
        }
    }
}

File: src/lec15/Lec15Map.kt
package lec15

fun main() {
    val map = mutableMapOf<Int, String>()

    map.put(5, "D") // 자바 처럼도 사용 가능하다
    map[3] = "A"
    map[4] = "B"

    // 맵 생성과 동시에 값을 입력할 때는 to를 사용
    val map2 = mapOf(1 to "A", 2 to "B")    // 불변


    // key만 가져오기
    for (key in map.keys) {
        println(key)        // key

        println(map.get(key))       // value
        println(map.getValue(key))  // value
        println(map[key])           // value
    }

    // key-value 쌍으로 가져오기
    for ((key, value) in map2.entries) {
        println("$key, $value")
    }

}